api/src/webhooks.ts

```diff
@@ -6,15 +6,31 @@ import { Webhook, WebhookHeader } from './types';
 import { WebhooksService } from './services';
 import { getSchema } from './utils/get-schema';
 import { ActionHandler } from '@directus/shared/types';
+import { getMessenger } from './messenger';
 
 let registered: { event: string; handler: ActionHandler }[] = [];
 
-export async function register(): Promise<void> {
+export async function init(): Promise<void> {
+	await register();
+	const messenger = getMessenger();
+
+	messenger.subscribe('webhooks', (event) => {
+		if (event.type === 'reload') {
+			reload();
+		}
+	});
+}
```

Comment:
All of these publish/subscribe operations are "async" but can't really be "awaited" (you can't know how many subscribers there are when you publish something, so you can't await all subscribers to finish their callback function). We should figure out some way to make sure that unexpected errors / messages / reload failures are handled